File: controller/comments/get_item.php
<?php
namespace Core;
use Controller\Main;
use Exception;

if (!defined('__API_GOOSE__')) exit();

/**
 * get comment
 *
 * @var Goose|Connect $this
 */

try
{
  // check and set srl
  if (($srl = (int)($this->params['srl'] ?? 0)) <= 0)
  {
    throw new Exception(Message::make('error.notFound', 'srl'));
  }

  // connect db
  $this->model->connect();

  // check access
  $token = Main::checkAccessItem($this, (object)[
    'table' => 'comments',
    'srl' => $srl,
    'useStrict' => true,
  ]);

  // set output
  $output = Main::item($this, (object)[
    'table' => 'comments',
    'srl' => $srl,
  ]);

  if ($output->data ?? false)
  {
    $ext_field = $this->get->ext_field ?? null;

    // get username
    if (Util::checkKeyInExtField('user_name', $ext_field))
    {
      $user = $this->model->getItem((object)[
        'table' => 'users',
        'field' => 'name',
        'where' => 'srl='.(int)$output->data->user_srl,
      ]);
      $output->data->user_name = $user->data->name;
    }
  }

  // set token
  if ($token) $output->_token = $token->jwt;

  // disconnect db
  $this->model->disconnect();

  // output data
  return Output::result($output);
}
catch (Exception $e)
{
  if (isset($this->model)) $this->model->disconnect();
  return Error::result($e->getMessage(), $e->getCode());
}

File: src/endpoints/category/get_item.py
from . import __types__ as types
from src import output
from src.libs.db import DB, Table

async def get_item(params: types.GetItem, _db: DB = None):

    # set values
    result = None

    # connect db
    if _db: db = _db
    else: db = DB().connect()

    try:
        # set fields
        fields = params.fields.split(',') if params.fields else None

        # set data
        data = db.get_item(
            table_name = Table.CATEGORY.value,
            fields = fields,
            where = [ f'and srl = {params.srl}' ],
        )

        # set result
        result = output.success({
            'message': 'Complete get Category item.',
            'data': data,
        })
    except Exception as e:
        result = output.exc(e)
    finally:
        if not _db: db.disconnect()
        return result
